src/libteam/0005-libteam-Add-warm_reboot-mode.patch

```diff
@@ -0,0 +1,354 @@
+diff --git a/libteam/ifinfo.c b/libteam/ifinfo.c
+index 72155ae..266cc21 100644
+--- a/libteam/ifinfo.c
++++ b/libteam/ifinfo.c
+@@ -58,6 +58,7 @@ struct team_ifinfo {
+ 	char			ifname[IFNAMSIZ];
+ 	uint32_t		master_ifindex;
+ 	bool			admin_state;
++        bool                    orig_addr_updated; /* FIXME: Check this. I think we don't need this flag */
+ #define MAX_PHYS_PORT_ID_LEN 32
+ 	char			phys_port_id[MAX_PHYS_PORT_ID_LEN];
+ 	size_t			phys_port_id_len;
+@@ -105,15 +106,17 @@ static void update_hwaddr(struct team_ifinfo *ifinfo, struct rtnl_link *link)
+ 	hwaddr_len = nl_addr_get_len(nl_addr);
+ 	if (ifinfo->hwaddr_len != hwaddr_len) {
+ 		ifinfo->hwaddr_len = hwaddr_len;
+-		if (!ifinfo->master_ifindex)
++		if (!ifinfo->orig_addr_updated)
+ 			ifinfo->orig_hwaddr_len = hwaddr_len;
+ 		set_changed(ifinfo, CHANGED_HWADDR_LEN);
+ 	}
+ 	hwaddr = nl_addr_get_binary_addr(nl_addr);
+ 	if (memcmp(ifinfo->hwaddr, hwaddr, hwaddr_len)) {
+ 		memcpy(ifinfo->hwaddr, hwaddr, hwaddr_len);
+-		if (!ifinfo->master_ifindex)
++		if (!ifinfo->orig_addr_updated) {
+ 			memcpy(ifinfo->orig_hwaddr, hwaddr, hwaddr_len);
++                        ifinfo->orig_addr_updated = true;
++                }
+ 		set_changed(ifinfo, CHANGED_HWADDR);
+ 	}
+ }
+diff --git a/teamd/teamd.c b/teamd/teamd.c
+index c987333..37c2fb3 100644
+--- a/teamd/teamd.c
++++ b/teamd/teamd.c
+@@ -116,7 +116,9 @@ static void print_help(const struct teamd_context *ctx) {
+             "    -D --dbus-enable         Enable D-Bus interface\n"
+             "    -Z --zmq-enable=ADDRESS  Enable ZeroMQ interface\n"
+             "    -U --usock-enable        Enable UNIX domain socket interface\n"
+-            "    -u --usock-disable       Disable UNIX domain socket interface\n",
++            "    -u --usock-disable       Disable UNIX domain socket interface\n"
++            "    -w --warm-reboot         Warm-reboot startup mode\n"
++            "    -L --lacp-directory      Directory for saving lacp pdu dumps\n",
+             ctx->argv0);
+ 	printf("Available runners: ");
+ 	for (i = 0; i < TEAMD_RUNNER_LIST_SIZE; i++) {
+@@ -149,10 +151,12 @@ static int parse_command_line(struct teamd_context *ctx,
+ 		{ "zmq-enable",		required_argument,	NULL, 'Z' },
+ 		{ "usock-enable",	no_argument,		NULL, 'U' },
+ 		{ "usock-disable",	no_argument,		NULL, 'u' },
++		{ "warm-reboot",        no_argument,		NULL, 'w' },
++		{ "lacp-directory",     required_argument,	NULL, 'L' },
+ 		{ NULL, 0, NULL, 0 }
+ 	};
+ 
+-	while ((opt = getopt_long(argc, argv, "hdkevf:c:p:groNt:nDZ:Uu",
++	while ((opt = getopt_long(argc, argv, "hdkevf:c:p:groNt:nDZ:UuwL:",
+ 				  long_options, NULL)) >= 0) {
+ 
+ 		switch(opt) {
+@@ -230,6 +234,12 @@ static int parse_command_line(struct teamd_context *ctx,
+ 		case 'u':
+ 			ctx->usock.enabled = false;
+ 			break;
++		case 'w':
++			ctx->warm_reboot = true;
++			break;
++		case 'L':
++			ctx->lacp_directory = strdup(optarg);
```

Comment:
can you make sure the directory exist here? we do not want to get error msg when we try to dump the lacp pdu to disk, if directory does not exist, we'd like to know the error msg early.

Reply:
What to do after the error message? exit?

Reply:
Fixed. Just error message, no exit